src/macaron/database/db_custom_types.py

```diff
@@ -60,3 +60,39 @@ def process_result_value(self, value: None | str, dialect: Any) -> None | dateti
         if result.tzinfo:
             return result
         return result.astimezone(RFC3339DateTime._host_tzinfo)
+
+
+class DBJsonDict(TypeDecorator):  # pylint: disable=W0223
+    """SQLAlchemy column type to serialize dictionaries."""
+
+    # It is stored in the database as a json value.
+    impl = JSON
+
+    # To prevent Sphinx from rendering the docstrings for `cache_ok`, make this docstring private.
+    #: :meta private:
+    cache_ok = True
+
+    def process_bind_param(self, value: None | dict, dialect: Any) -> None | dict:
+        """Process when storing a dict object to the SQLite db.
+
+        value: None | dict
+            The value being stored
+        """
+        if value is None:
+            return None
+        if not isinstance(value, dict):
```

Comment:
Because the type for `value` is `None | dict`, would this check be unnecessary as we have checked for `None` before?

Reply:
See commit [7178e3d](https://github.com/oracle/macaron/pull/797/commits/7178e3d575a1086abf740cbab1025abc6c1dfbc7)